src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -2515,6 +2565,11 @@ - (NSArray *)_getAttributesToReportWithReportedValues:(NSArray<NSDictionary<NSSt
             // when our cached value changes and no expected value exists.
             if (readCacheValueChanged && !expectedValue) {
                 shouldReportAttribute = YES;
+
+                _deviceConfigurationChanged = [self _isAttributeAffectingDeviceConfigurationChanged:attributePath];
```

Comment:
This line looks wrong to me, both in terms of what it does and in terms of where it is located.  What is the logic we are trying to implement here?

Please write tests that would have caught at least the "what it does" bug; the other might be harder to test.